### PR TITLE
Add tour items as a dropdown to masterbar

### LINKED ISSUE #13: Add the tour selector to the Masterbar
If the user case is logged-in users, we can add the available tours as a dropdown in the masterbar.

### PR DESCRIPTION
**Pull request by trymebytes**
In this PR we add a dropdown to the masterbar that shows a list of available tours.

tourmasterbar

Fixes #13

### CODE REVIEW

File: tour.php [hunk 1]
```diff
@@ -668,3 +649,33 @@ function tour_available_tours_init() {
 	) );
 }
 add_action( 'init', 'tour_available_tours_init' );
+
+function add_tour_item_to_masterbar() {
+    global $wp_admin_bar;
+	if ( empty( apply_filters( 'tour_list', array() ) ) ) {
+		return '<p>' . esc_html__('There are no tours available.', 'tour') . '</p>';
```
Comment (akirk) [inline tour.php]: Shouldn't this just return nothing or a boolean?

File: tour.php [hunk 2]
```diff
@@ -668,3 +649,33 @@ function tour_available_tours_init() {
...
+	}
+	$tours = apply_filters( 'tour_list', array() );
+	$wp_admin_bar->add_menu(
+		array(
+			'id'    => 'tour-list',
+			'title' => esc_html__('Available Tours', 'tour'),
```
Comment (akirk) [inline tour.php]:
```suggestion
			'title' => esc_html__('Tours', 'tour'),
```

**Review by akirk — COMMENTED** [review]
Nice! On certain pages on my site the menu is empty because the trigger cannot be found. This should also remove the Masterbar entry.

File: assets/js/tour-admin.js [hunk 3]
```diff
@@ -206,3 +206,40 @@ document.addEventListener('keyup', function(event) {
...
+			} );
+			for ( let i = 0; i < tourListItems.length; i++ ) {
+				let stringSplit = tourListItems[i].id.split('-');
+				let tourId = stringSplit[stringSplit.length - 1];
+				
+				if (  ! tourIdsOnPage.includes( tourId ) ) {
+					document.querySelector('#wp-admin-bar-tour-' + tourId).remove();
+				}
+			}
+	}
```
Comment (amieiro) [inline assets/js/tour-admin.js]: Currently, if the page doesn't have any tour, it shows a weird empty dropdown. I suggest two possible actions:
- Remove the "Available tours" element from the admin bar.
- Add some text, maybe in orange (warning color), saying that in this page we don't have any tour: There are no tours available on this page.

image

**Comment by akirk** [comment]
Nice! In my brief test I noticed that it doesn't have the same "restart" behavior as the shortcode or Gutenberg block, i.e. when a tour is already finished, it would start from 0.

**Comment by amieiro** [comment]
I have 1 test in the backend and 1 in the frontend. 

In the backend the listing works fine, but not in the frontend, because it shows both tests. See the screenshots.

image

image

Another weird thing happens in the frontend, in a page without a tour, is that it shows a tour in the listing.

image

**Review by akirk — APPROVED** [review]
Works fine for me

**Pull request merged**